### DISCUSSION TITLE
404: The requested URL was not found on this server for shortened URLs

### DISCUSSION THREAD

**Discussion by tcreek** [YOURLS/YOURLS#3809] [Q&A] [answered]
I have looked at both this Discussion, and the Issues looking for the issue, but no luck

This is with Apache, and Debian 12
I am using the standard .htaccess 

```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS

```

sites-available http host config file. The bottom four lines were added by Let's Encrypt
```

<VirtualHost *:80>

    ServerName lindo.ws
    ServerAlias www.lindo.ws
    DocumentRoot /var/www/html/lindo.ws
    ErrorLog ${APACHE_LOG_DIR}/lindo.ws.error.log
    CustomLog ${APACHE_LOG_DIR}/lindo.ws.access.log combined
    Options Indexes FollowSymLinks


RewriteEngine on
RewriteCond %{SERVER_NAME} =www.lindo.ws [OR]
RewriteCond %{SERVER_NAME} =lindo.ws
RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
</VirtualHost>

```

And the SSL conf (also edited by Let's Encrypt)

```
<IfModule mod_ssl.c>
<VirtualHost *:443>

    ServerName lindo.ws
    ServerAlias www.lindo.ws
    DocumentRoot /var/www/html/lindo.ws
    ErrorLog ${APACHE_LOG_DIR}/lindo.ws.error.log
    CustomLog ${APACHE_LOG_DIR}/lindo.ws.access.log combined
    RewriteEngine on


# Some rewrite rules in this file were disabled on your HTTPS site,
# because they have the potential to create redirection loops.

# RewriteCond %{SERVER_NAME} =www.lindo.ws [OR]
# RewriteCond %{SERVER_NAME} =lindo.ws
# RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]

SSLCertificateFile /etc/letsencrypt/live/lindows.org/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/lindows.org/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

I tried uncommenting those changes, but still not working out. Instead I get "The page isn’t redirecting properly "

Thanks!

**Comment by tcreek** [comment]
bump?

**Reply by dgw** [reply]
This error is [the top entry on the Frequent Issues page](https://yourls.org/docs/guide/troubleshooting/frequent-issues#404-instead-of-redirection).

Mixing rewrite rules in different contexts seems a little risky based on a bit of research just now, e.g. https://serverfault.com/q/173910/13440 talks about Directory vs. VirtualHost vs. .htaccess and different inheritance modes.

I don't use Apache myself, but I can tell you that this issue almost certainly comes from how your specific instance is configured. Personally I would either put all of the rewrites for a given domain in the conf file _or_ the .htaccess, but never spread across both.

**Comment by tcreek** [comment, marked as answer]
This may have been the same error, but the troubleshooting steps and documentation was not helpful.
I am clueless on ll this redirect stuff, and most of of all the other settings, but your server fault link was slightly useful, though did not solve the issue.

There was a mention of `<Directory>` in the site configuration which I did not have. Doing some research on that I was able to come up with a configuration which made it work.

In the virtual site configuration, and site SSL configuration, I needed to add:


```
<Directory /var/www/html/lindo.ws>
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
```

Maybe you can add something like that to the documentation.

Thanks for the tips!
